packages/core/src/messagingProtocols/mls/E2EIdentityService/E2EIServiceExternal.ts

```diff
@@ -78,21 +68,49 @@ export class E2EIServiceExternal {
     return this.coreCryptoClient.e2eiIsEnabled(ciphersuite);
   }
 
-  public async getUsersIdentities(groupId: string, userIds: QualifiedId[]) {
-    return this.coreCryptoClient.getUserIdentities(
+  public async getUsersIdentities(groupId: string, userIds: QualifiedId[]): Promise<Map<string, DeviceIdentity[]>> {
+    // We keep track of the ID we give to CoreCrypto in order to map it back to regular userIds afterwards
+    const hashedIdMap = new Map(userIds.map(userId => [uuidTobase64url(userId.id).asString, userId]));
+
+    const userIdentities = await this.coreCryptoClient.getUserIdentities(
       Decoder.fromBase64(groupId).asBytes,
-      userIds.map(userId => uuidTobase64url(userId.id).asString),
+      Array.from(hashedIdMap.keys()),
     );
+
+    const mappedUserIdentities = new Map();
+    for (const [base64Id, identities] of userIdentities) {
+      // remapping coreCrypto user ids to regular userIds
+      const userId = hashedIdMap.get(base64Id);
+      if (userId) {
+        mappedUserIdentities.set(
+          userId.id,
+          identities.map(identity => ({
+            ...identity,
+            deviceId: parseFullQualifiedClientId((identity as any).client_id).client,
+          })),
+        );
+      }
+    }
+
+    return mappedUserIdentities;
   }
 
   // Returns devices e2ei certificates
-  public async getUserDeviceEntities(
+  public async getDevicesIdentities(
     groupId: string,
     userClientsMap: Record<string, QualifiedId>,
-  ): Promise<WireIdentity[]> {
+  ): Promise<DeviceIdentity[]> {
     const clientIds = Object.entries(userClientsMap).map(
       ([clientId, userId]) => getE2EIClientId(clientId, userId.id, userId.domain).asBytes,
     );
-    return this.coreCryptoClient.getDeviceIdentities(Decoder.fromBase64(groupId).asBytes, clientIds);
+    const deviceIdentities = await this.coreCryptoClient.getDeviceIdentities(
+      Decoder.fromBase64(groupId).asBytes,
+      clientIds,
+    );
+
+    return deviceIdentities.map(identity => ({
+      ...identity,
+      deviceId: parseFullQualifiedClientId((identity as any).client_id).client,
```

Comment:
The idea here is to parse the `clientId` from coreCrypto and convert it to a format that the consumer (namely the webapp) can understand